krabcake/rs_hello/src/lib.rs

```diff
@@ -714,20 +741,18 @@ pub extern "C" fn rs_trace_store(
             );
         }
 
-        if shadow_addr != 0 {
-            // A non-trivial shadow on the address means this was a
-            // *tagged* pointer; need to confirm this access is legal
-            // (and update the stack to reflect a use via this tag).
-            check_use_1(addr, shadow_addr);
-        }
+        // A non-trivial shadow on the address means this was a
+        // *tagged* pointer; need to confirm this access is legal
+        // (and update the stack to reflect a use via this tag).
+        check_use_1(addr, shadow_addr);
```

Comment:
I think this might be what is triggering the panic @pnkfelix

Reply:
Bryan and I discussed this a bit outside github; we agreed that encoding the bottom tag as vgaddr  0 is *not* the right approach. A value with associated shadow metadata of 0 is meant to represent a value that has no extra tag at all (which is not the same as the bottom tag). So the right approach is to allocate different (non-zero) value to represent bottom.